private/pkg/git/git.go

```diff
@@ -135,3 +152,161 @@ type ListFilesAndUnstagedFilesOptions struct {
 	// is being applied to.
 	IgnorePathRegexps []*regexp.Regexp
 }
+
+// Remote represents a Git remote and provides associated metadata.
+type Remote interface {
+	// Name of the remote (e.g. "origin")
+	Name() string
+	// HEADBranch is the name of the HEAD branch of the remote.
+	HEADBranch() string
+	// Hostname is the host name parsed from the remote URL. If the remote is an unknown
+	// kind, then this may be an empty string.
+	Hostname() string
+	// RepositoryPath is the path to the repository based on the remote URL. If the remote
+	// is an unknown kind, then this may be an empty string.
+	RepositoryPath() string
+	// SourceControlURL makes the best effort to construct a user-facing source control url
+	// given a commit sha string based on the remoteKind and available hostname and
+	// repository path information.
+	//
+	// If the remote hostname contains bitbucket (e.g. bitbucket.mycompany.com or bitbucket.org),
+	// then it uses the route /commits for the git commit sha.
+	//
+	// If the remote hostname contains github (e.g. github.mycompany.com or github.com) or gitlab
+	// (e.g. gitlab.mycompany.com or gitlab.com) then it uses the route /commit for the git
```

Comment:
I think it would be clearer if you documented the full route returned, like:

`https://<hostname><repository-path>/commit/<git-commit-sha>`


(and same for bitbucket)